src/stmt.cpp

```diff
@@ -1133,8 +1133,17 @@ void ForStmt::EmitCode(FunctionEmitContext *ctx) const {
     ctx->RestoreContinuedLanes();
     ctx->ClearBreakLanes();
 
-    if (step)
+    if (step) {
         step->EmitCode(ctx);
+        for (auto &inst : *bstep) {
+            // If the instruction is a user of phi node induction variable, set no wrap attributes.
```

Comment:
The patch is working without the check for the phi node definition. Should we add it anyway?

Reply:
This is the wrong patch, we need to add only `NoSignedWrap` flag, only to operation with signed integers, and we need to do that for all Unary and Binary operations.